pkg/controllerhelpers/core.go

```diff
@@ -68,8 +77,15 @@ func aggregateStatusConditionInfo(conditions []metav1.Condition) (string, string
 		}
 	}
 
-	// TODO: Strip aggregated reasons and messages to fit into API limits.
-	return strings.Join(reasons, ","), strings.Join(messages, "\n")
+	return truncateIfNecessary(strings.Join(reasons, ","), maxReasonLength),
+		truncateIfNecessary(strings.Join(messages, "\n"), maxMessageLength)
```

Comment:
did you consider https://github.com/scylladb/scylla-operator/issues/1075#issuecomment-1286829019? We're losing some information here so I'd lean towards something similar to what was proposed in that comment

Reply:
That makes sense - PTAL now. I implemented it by omitting reasons/messages that we cannot fit into the aggregated string and indicating the number of omitted elements, as suggested.